pkg/sql/executor.go

```diff
@@ -1104,6 +1109,10 @@ func runWithAutoRetry(
 				e.TxnAbortCount.Inc(1)
 				txnState.mu.txn.CleanupOnError(session.Ctx(), err)
 				txnState.resetStateAndTxn(Aborted)
+				// autoCommit always leaves the transaction in NoTxn.
+				if autoCommit {
+					txnState.resetStateAndTxn(NoTxn)
```

Comment:
Is it useful/desirable to call resetStateAndTxn two times in a row?  Can't you combine them and use a if/else here somehow?